index.js

```diff
@@ -1,70 +1,68 @@
-let bookList = JSON.parse(localStorage.getItem('books')) || [];
-
-function removeBook(element, title, author) {
-  element.remove();
-  bookList = bookList.filter((book) => book.title !== title || book.author !== author);
-  localStorage.setItem('books', JSON.stringify(bookList));
-}
-
-function addBook(title, author) {
-  for (let i = 0; i < bookList.length; i += 1) {
-    if (bookList[i].title === title) {
-      alert('Book is already exist');
-      removeBook(title, author);
-    }
+class BookLibrary {
+  constructor() {
+    this.booksCollection = document.querySelector('.books');
+    this.bookList = JSON.parse(localStorage.getItem('books')) || [];
+    this.bookList.forEach((book) => {
+      const div = document.createElement('div');
+      div.innerHTML = `
+        <h2>${book.title}</h2>
+        <p>${book.author}</p>
+      `;
+
+      const removeBtn = document.createElement('button');
+      removeBtn.innerText = 'Remove';
+      removeBtn.addEventListener('click', () => {
+        this.removeBook(div, book.title, book.author);
+      });
+
+      div.appendChild(removeBtn);
+
+      this.booksCollection.appendChild(div);
+    });
   }
 
-  bookList.push({ title, author });
-
-  localStorage.setItem('books', JSON.stringify(bookList));
-}
+  addBook(title, author) {
+    for (let i = 0; i < this.bookList.length; i += 1) {
+      if (this.bookList[i].title === title) {
+        alert('this book is already exist');
+        this.removeBook(title, author);
+      }
+    }
+    this.bookList.push({ title, author });
+    localStorage.setItem('books', JSON.stringify(this.bookList));
+    const div = document.createElement('div');
+    div.innerHTML = `
+    <h2>${title}</h2>
+    <p>${author}</p>
+    `;
 
-const booksCollection = document.querySelector('.books');
+    const removeBtn = document.createElement('button');
+    removeBtn.innerText = 'Remove';
+    removeBtn.addEventListener('click', () => {
+      this.removeBook(div, title, author);
+    });
 
-bookList.forEach((book) => {
-  const div = document.createElement('div');
-  div.innerHTML = `
-    <h2>${book.title}</h2>
-    <p>${book.author}</p>
-  `;
+    div.appendChild(removeBtn);
 
-  const removeBtn = document.createElement('button');
-  removeBtn.innerText = 'Remove';
-  removeBtn.addEventListener('click', () => {
-    removeBook(div, book.title, book.author);
-  });
+    this.booksCollection.appendChild(div);
+  }
 
-  div.appendChild(removeBtn);
+  removeBook(element, title, author) {
+    element.remove();
```

Comment:
- [ ] Whenever I am trying to add another book with the same title I get an alert that the same title already exists and I cannot add it, that is fine but with it, I am getting an error in the console which you should fix it
![Screenshot from 2021-11-25 19-24-06](https://user-images.githubusercontent.com/62705947/143487954-2b007481-d163-4189-aae4-be5702c5bbf3.png)
.

Reply:
- [ ] I still see this  `console.log` error, kindly fix this before submitting another review request 😄
    ![imagen](https://user-images.githubusercontent.com/60789317/143493172-75aad4d0-9ad8-48b3-ad31-e9c223a0e7b2.png)

Reply:
No more console error, thank you